src/core/curry.js

```diff
@@ -5,12 +5,15 @@ const isFunction = require('./isFunction')
 
 function applyCurry(fn, arg) {
   if(!isFunction(fn)) { return fn }
+
   return fn.length > 1 ? fn.bind(null, arg) : fn.call(null, arg)
 }
 
 // curry : ((a, b, c) -> d) -> a -> b -> c -> d
 function curry(fn) {
-  return function(...xs) {
+  if(fn['@@crocks/curried']) { return fn }
```

Comment:
How about we create a constant for the signal and use a multiline `if` (should probably add that to lint soon)
```js
const CURRY_SYMB =
  '@@crocks/curried'

if(fn[CURRY_SYMB]) {
  return fn
}
```